synapse/types/handlers/sliding_sync.py

```diff
@@ -158,6 +158,7 @@ class RoomResult:
                 name changes to mark the room as unread and bump it to the top. For
                 encrypted rooms, we just have to consider any activity as a bump because we
                 can't see the content and the client has to figure it out for themselves.
+                This is only included if there has been a change.
```

Comment:
We seem to go against this statement if we can find a `bump_stamp` within the timeline events. Also for any non-join room. (because it's easy and doesn't require more work)

That behavior isn't necessarily bad/wrong but doesn't agree with what we say we should be doing here.

To match the behavior, we could simply update this to be:

```suggestion
                This may not be included if there hasn't been a change.
```